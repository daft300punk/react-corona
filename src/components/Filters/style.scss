.container {
  display: flex;
  flex-direction: column;
  position: absolute;
  left: 5%;
  top: calc(50% - 40px);
  transform: translateY(calc(-50%));
  background-color: rgba(255, 255, 255, .01);
  border: 1px solid rgba(255, 255, 255, .05);
  border-radius: 8px;
  font-family: sans-serif;
  color: gray;
  opacity: 0.25;
  padding: 24px;
  width: 15%;
  min-width: 250px;

  &:hover {
    opacity: 1;
  }

  p {
    font-size: 13px;
  }

  h2 {
    color: white;
    padding: 0;
    margin: 0;
  }

  h4 {
    padding: 0;
    margin: 0;
    color: white;
  }

  ul {
    list-style-type: none;
    display: flex;
    flex-direction: column;
    margin-top: 8px;
  }

  ul li {
    margin-left: -24px;
    cursor: pointer;
    align-self: flex-start;
    padding: 4px 8px 4px 8px;
    font-size: 13px;
  }

  li.active {
    background-color: rgba(255, 255, 255, .05);
    border-radius: 10px;
    border: 1px solid #8cc543;
  }
}

@media only screen 
and (max-width: 1024px) {
  .container {
    display: none;
  }
}
