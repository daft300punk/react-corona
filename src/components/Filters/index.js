import React from 'react';
import './style.scss';
import { categories } from '../../constants/categoriesAndFilters';
import PropTypes from 'prop-types';

/*
  The filters component hanldes filtering the data of a particular
  category. 4 filters are available as of now. This can be changed as
  needed.
*/
const Filters = ({
  selectedCategory,
  selectedFilter,
  changeSelectedFilter
}) => {
  const filters = categories[selectedCategory].filters;
  return (
    <div className="container">
      <h2>{selectedCategory.toUpperCase()}</h2>
      <p>{categories[selectedCategory].text}</p>
      <h4>Select Filters</h4>
      <ul>
        {
          filters.map((filter, i) => {
            let className = '';
            if (selectedFilter === filter.toLowerCase())
              className += 'active';
            return (
              <li
                key={i}
                onClick={() => changeSelectedFilter(filters[i])}
                className={className}
              >
                {filter}
              </li>
            );
          })
        }
      </ul>
    </div>
  );
};

Filters.propTypes = {
  selectedCategory: PropTypes.string,
  selectedFilter: PropTypes.string,
  changeSelectedFilter: PropTypes.func
};

export default Filters;
