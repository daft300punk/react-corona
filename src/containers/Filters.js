import React from 'react';
import { connect } from 'react-redux';
import Filters from '../components/Filters';
import { changeSelectedFilter } from '../actions';
import PropTypes from 'prop-types';

/*
  See Fitlers component to get the idea. @selectedCategory
  is the currently selected category out of 5 available.
  @changeSelectedFilter is a bound dispathcer that gets
  called when the user clicks on a filter other than current.
  @selectedFilter is the currently selected filter from the list
  of available 4.
*/
const FiltersContainer = ({
  selectedCategory,
  changeSelectedFilter,
  selectedFilter
}) => {
  return (
    <Filters
      selectedCategory={selectedCategory}
      changeSelectedFilter={changeSelectedFilter}
      selectedFilter={selectedFilter}
    />
  );
};

const mapStateToProps = (state) => ({
  selectedCategory: state.selectedOptions.category,
  selectedFilter: state.selectedOptions.filter
});

const mapDispatchToProps = (dispatch) => ({
  changeSelectedFilter: (filter) => { dispatch(changeSelectedFilter(filter)); }
});

FiltersContainer.propTypes = {
  selectedCategory: PropTypes.string,
  selectedFilter: PropTypes.string,
  changeSelectedFilter: PropTypes.func
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(FiltersContainer);
