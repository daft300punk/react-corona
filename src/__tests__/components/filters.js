import React from 'react';
import renderer from 'react-test-renderer';

import Filters from '../../components/Filters';

describe('Filters Component', () => {
  it('renders Filters correctly', () => {
    const component = renderer.create(
      <Filters
        selectedCategory={'all activity'}
        changeSelectedCategory={() => { }}
        selectedFilter={'all time'}
      />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});