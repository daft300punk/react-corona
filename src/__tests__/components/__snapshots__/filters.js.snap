exports[`Filters Component renders Filters correctly 1`] = `
<div
  className="container">
  <h2>
    ALL ACTIVITY
  </h2>
  <p>
    1. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.
  </p>
  <h4>
    Select Filters
  </h4>
  <ul>
    <li
      className="active"
      onClick={[Function]}>
      All Time
    </li>
    <li
      className=""
      onClick={[Function]}>
      Last Week
    </li>
    <li
      className=""
      onClick={[Function]}>
      Last Month
    </li>
    <li
      className=""
      onClick={[Function]}>
      Last Year
    </li>
  </ul>
</div>
`;
